### PR TITLE
feat: Sanitize user entered 2FA code before submitting (STARGATE-1965)

### PR DESCRIPTION
**Pull request by lukewhyte**
I left a similar comment in the ticket but, maybe in the future we should consider validating this code and displaying an error message instead of just sanitizing and letting her rip?

### CODE REVIEW

File: packages/pci-sdk-iframe/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
 		"deploy-to-s3": "node scripts/deploy-to-s3.js",
 		"prebuild": "webpack",
 		"prepublishOnly": "npm run build && npm run create-s3-artifacts && npm run deploy-to-s3",
-		"start": "react-scripts start",
+		"start": "BROWSER=none react-scripts start",
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/package.json]: 🤔 Not sure if we want this.

Maybe you can add this to your own .env file

File: packages/pci-sdk-iframe/src/apiClient/index.ts [hunk 2]
```diff
@@ -43,10 +43,12 @@ interface IVerify2FACodeResponse {
 }
 
 export async function verify2FACode(secret: string, verificationId: string): Promise<IVerify2FACodeResponse> {
+	const santizedSecret = formatterService.sanitize2FACode(secret);
+
 	return fetch(`${BASE_URL}v1/verifications/${verificationId}/finish`, {
 		method: 'POST',
 		headers,
-		body: JSON.stringify({ secret }),
+		body: JSON.stringify({ santizedSecret }),
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/src/apiClient/index.ts]: Is this even working? ¬¬

you are sending a totally new object with this structure:

```ts
{
  "santizedSecret" : "whatever"
}
```

while the server expects 

```ts
{
  "secret" :  "whatever"
}
```

This is worrying.

I know that being a matter of integration with the API we do not have automatic tests. But you should have tested it manually or running cypress E2E. What happened? Didn't you run the tests or the tests didn't detect the error?

**Review by IagoLast — CHANGES_REQUESTED** [review]

File: packages/pci-sdk-iframe/src/apiClient/index.ts (hunk 2 above)
Comment (lukewhyte) [reply packages/pci-sdk-iframe/src/apiClient/index.ts]: Bah, you are right. On first commit I was sanitizing the code in the hook and then, just before logging off for the day I hastily moved the logic to the API. I reran the unit tests but not the cypress tests... maybe I should see why it wasn't caught by the unit tests.

**Comment by lukewhyte** [comment]
@IagoLast I made a number of changes to our unit tests to improve the robustness of our testing. Lets discuss tomorrow but, in a nutshell:

 - We're now spying directly on `fetch` during the `showCardData` tests and making sure that the write body always being sent in the test. We're also using `toHaveBeenNthCalledWith` to make sure requests happen in the correct order
 - I added a number of additional tests during `showCardData` to make sure we cover each possible request and error

**Review by IagoLast — APPROVED** [review]

**Pull request merged**